add public getters to WitnessCS fields to permit arecibo refactor

jobez opened the pull request:
Per this discussion: https://github.com/lurk-lab/arecibo/issues/77#issuecomment-1787358390 

This change makes the WitnessCS fields public, enabling [arecibo's SatisfyingAssignment](https://github.com/lurk-lab/arecibo/blob/dev/src/bellpepper/solver.rs#L9) to be reimplemented as a type alias.

huitseeker reviewed (commented):
Making this public means making those fields mutable as well as readable. Are there any instances in arecibo where it's necessary to mutate?

If not, would a public getter suffice ?

If a public getter would be enough, please note the [API conventions on naming getters](https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter).

jobez commented:
> Making this public means making those fields mutable as well as readable. Are there any instances in arecibo where it's necessary to mutate?
> 
> If not, would a public getter suffice ?
> 
> If a public getter would be enough, please note the [API conventions on naming getters](https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter).

I removed these getters as they weren't being used anywhere 

https://github.com/lurk-lab/bellpepper/blob/ac5f2bf388b76c06d39e5788b688f5284d798d7f/crates/bellpepper/src/util_cs/witness_cs.rs#L170-L178

And added `_assignment` based getters

huitseeker commented on crates/bellpepper/src/util_cs/witness_cs.rs:
```diff
@@ -51,6 +51,19 @@ where
+where
+    Scalar: PrimeField,
+{
+    pub fn input_assignment(&self) -> &Vec<Scalar> {
+        &self.input_assignment
+    }
+
+    pub fn aux_assignment(&self) -> &Vec<Scalar> {
+        &self.aux_assignment
+    }
```
You probably want to return a `&[Scalar]` for each of these two functions.

huitseeker reviewed (commented):
I have just the one comment, otherwise this looks great!

huitseeker commented on crates/bellpepper/src/util_cs/witness_cs.rs:
```diff
@@ -166,13 +179,3 @@ where
     }
 }
-
-impl<Scalar: PrimeField> WitnessCS<Scalar> {
-    pub fn scalar_inputs(&self) -> Vec<Scalar> {
-        self.input_assignment.clone()
-    }
-
-    pub fn scalar_aux(&self) -> Vec<Scalar> {
-        self.aux_assignment.clone()
-    }
-}
```
If we're removing those, I'd rather see PRs in neptune to check we can switch to the new ones.

huitseeker commented on crates/bellpepper/src/util_cs/witness_cs.rs:
```diff
@@ -167,6 +180,10 @@ where
     }
 }
 
+#[deprecated(
```
I'd rather have the deprecation done per-function, which will allow the developer in a downstream crate to get an error messages directly relevant to the method they are using. IOW, it will be good that the developer using `scalar_inputs` would not have to ask themselves, what `aux_assignment` does.

jobez replied:
Makes sense!

huitseeker reviewed (commented):
One comment left, otherwise this is good to go.

huitseeker reviewed (approved):
Tip top! thanks for your patience in getting through rounds of this!
